_source/_posts/2021-10-16-full-stack-java.md

```diff
@@ -0,0 +1,693 @@
+
+If you haven't heard of JHipster, boy do I have a treat for you! JHipster started as a Yeoman application generator back in 2013 and has grown to become a development platform. It allows you to quickly generate, develop, and deploy modern web apps and microservice architectures. Today, I'll show you how to build a Flickr clone with JHipster and lock it down with OAuth and OpenID Connect (OIDC).
+
+**Prerequisites:**
+
+- [Node.js 14+](https://nodejs.org/)
+- [Java 11+](https://sdkman.io)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+## Full Stack Development with React and Spring Boot
+
+One of the easiest ways to get started with React is by using [Create React App](https://create-react-app.dev/) (CRA). You install it locally, then run `create-react-app <project>` to generate a React application with minimal dependencies. It uses webpack under-the-covers to build the project, launch a web server, and run its tests.
+
+Spring Boot has a similar tool, called [Spring Initializr](https://start.spring.io/). Spring Initializer is a bit different than CRA because it's a website (and API) that you can create applications with.
+
+Today, I'll show you how to build a Flickr clone with React and Spring Boot. However, I'm going to cheat. Rather than building everything using the aforementioned tools, I'm going to use JHipster. JHipster is an application generator that initially only supported Angular and Spring Boot. Now it supports Angular, React, and Vue for the frontend. JHipster also has support for [Kotlin, Micronaut, Quarkus, .NET, and Node.js](https://www.jhipster.tech/modules/official-blueprints/) on the backend.
+
+In this tutorial, we'll use React since it seems to be [the most popular](https://trends.google.com/trends/explore?q=angular,react,vuejs) frontend framework nowadays. 
+
+## Get Started with JHipster 7
+
+To get started with JHipster, you'll need a fast internet connection and Node.js installed. The project recommends you use the latest LTS (Long Term Support) version, which is 14.7.6 at the time of this writing. To run the app, you'll need to have Java 11 installed. If you have Git installed, JHipster will auto-commit your project after creating it. This will allow you to upgrade between versions.
+
+Run the following command to install JHipster:
+
+```shell
+npm i -g generator-jhipster@7
+```
+
+To create a full-stack app with JHipster, create a directory, and run `jhipster` in it:
+
+```shell
+mkdir full-stack-java
+cd full-stack-java
+jhipster
+```
+
+JHipster will prompt you for the type of application to create and what technologies you'd like to include. For this tutorial, make the following choices:
+
+| Question | Answer |
```

Comment:
let's maybe wrap this into ```bash  block

Reply:
It's a table now. I think it might look better if we just specified the defaults to override, but I had a hard time finding the words to express that.